### PR TITLE
First attempt at GH Action updates

### PR DESCRIPTION
**Pull request by karianna**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -38,11 +38,11 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```
Comment (d3r3kk) [inline .github/workflows/codeql-analysis.yml:41]: Latest [suggested version](https://github.com/actions/checkout#usage) is just v4. Should do... and will keep updating until v4 is no longer updated (and v5 becomes the main version to follow).

```suggestion
      uses: actions/checkout@v4
```

File: .github/workflows/early_access.yml [hunk 2]
```diff
@@ -12,13 +12,13 @@ jobs:
     if: startsWith(github.event.head_commit.message, '[maven-release-plugin]') != true
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```
Comment (d3r3kk) [inline .github/workflows/early_access.yml:15]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/early_access.yml [hunk 3]
```diff
@@ -12,13 +12,13 @@ jobs:
...
 
     - name: Set up JDK 11
-      uses: actions/setup-java@v2
+      uses: actions/setup-java@0ab4596768b603586c0de567f2430c30f5b0d2b0 # v3.13.0
```
Comment (d3r3kk) [inline .github/workflows/early_access.yml:18]: V2 is good. [See this article...](https://github.com/actions/setup-java#v2-vs-v1)

```suggestion
      uses: actions/setup-java@v2
```

File: .github/workflows/early_access.yml [hunk 4]
```diff
@@ -12,13 +12,13 @@ jobs:
...
       with:
         java-version: 11
-        distribution: 'adopt'
+        distribution: 'temurin'
```
Comment (d3r3kk) [inline .github/workflows/early_access.yml:21]: You meant Microsoft, right? 😁 

```suggestion
        distribution: 'microsoft'
```

File: .github/workflows/maven.yml [hunk 5]
```diff
@@ -21,13 +21,13 @@ jobs:
         java: [11, 17]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```
Comment (d3r3kk) [inline .github/workflows/maven.yml:24]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/maven.yml [hunk 6]
```diff
@@ -21,13 +21,13 @@ jobs:
...
 
     - name: Set up JDK ${{ matrix.java }}
-      uses: actions/setup-java@v2
+      uses: actions/setup-java@0ab4596768b603586c0de567f2430c30f5b0d2b0 # v3.13.0
```
Comment (d3r3kk) [inline .github/workflows/maven.yml:27]:
```suggestion
      uses: actions/setup-java@v2
```

File: .github/workflows/maven.yml [hunk 7]
```diff
@@ -21,13 +21,13 @@ jobs:
...
       with:
         java-version: ${{ matrix.java }}
-        distribution: 'adopt'
+        distribution: 'temurin'
```
Comment (d3r3kk) [inline .github/workflows/maven.yml:30]:
```suggestion
        distribution: 'microsoft'
```

File: .github/workflows/publish.yml [hunk 8]
```diff
@@ -10,23 +10,23 @@ jobs:
 
     steps:
       - name: Checkout project
-        uses: actions/checkout@v2
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
```
Comment (d3r3kk) [inline .github/workflows/publish.yml:13]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/publish.yml [hunk 9]
```diff
@@ -10,23 +10,23 @@ jobs:
...
         with:
           ref: main
           fetch-depth: 0
 
       - name: Cache Maven
-        uses: actions/cache@v2.1.4
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
```
Comment (d3r3kk) [inline .github/workflows/publish.yml:19]: Looks like  the [latest major version](https://github.com/actions/cache#restoring-and-saving-cache-using-a-single-action) to use for cache is v3.

```suggestion
        uses: actions/cache@v3
```

File: .github/workflows/publish.yml [hunk 10]
```diff
@@ -10,23 +10,23 @@ jobs:
...
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
 
       - name: Setup Java JDK
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@0ab4596768b603586c0de567f2430c30f5b0d2b0 # v3.13.0
```
Comment (d3r3kk) [inline .github/workflows/publish.yml:26]:
```suggestion
        uses: actions/setup-java@v2
```

File: .github/workflows/publish.yml [hunk 11]
```diff
@@ -10,23 +10,23 @@ jobs:
...
         with:
           java-version: '11'
-          distribution: 'adopt'
+          distribution: 'temurin'
```
Comment (d3r3kk) [inline .github/workflows/publish.yml:29]:
```suggestion
          distribution: 'microsoft'
```

**Review by d3r3kk — CHANGES_REQUESTED** [review]
Looks mostly good, but a few recommendations. Keeping to major versions of actions is best practice, AFAIK, and swapping for the Microsoft Build  of OpenJDK seems relevant 😁.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (kcpeppe) [reply .github/workflows/codeql-analysis.yml:41]: Can a link to an issue be provided? I'm trying to understand what is happening here in light of our upcoming changes to integrate the MS signing policy and our integration with jreleaser

Comment (karianna) [reply .github/workflows/codeql-analysis.yml:41]: This is just a GitHub Action version update for CI, it's separate to any jreleaser efforts.

File: .github/workflows/early_access.yml (hunk 3 above)
Comment (karianna) [reply .github/workflows/early_access.yml:18]: yeah but v3.13.0 is the latest :-)

File: .github/workflows/publish.yml (hunk 9 above)
Comment (karianna) [reply .github/workflows/publish.yml:19]: yes, it's 3.3.2, it's safer to use teh commit hash and a comment, dependabot knows what to do going forward

**Comment by karianna** [comment]
@d3r3kk FYI - the Better security practice is now to reference the hash commits of GH actions (as `vX` is an alias that can be updated under the hood) and comment on the side, Dependabot has been updated to cope with this new practice.

**Comment by d3r3kk** [comment]
> @d3r3kk FYI - the Better security practice is now to reference the hash commits of GH actions (as `vX` is an alias that can be updated under the hood) and comment on the side, Dependabot has been updated to cope with this new practice.

Oh interesting. Then I rescind all comments to that effect.

**Review by d3r3kk — APPROVED** [review]
With the updated info on best practices, I say 🚢 it.

**Pull request merged**